emu/src/io_registers.rs

```diff
@@ -0,0 +1,342 @@
+use crate::bitwise::Bits;
+
+pub enum IORegisterAccessControl {
+    Read,
+    Write,
+    ReadWrite,
+}
+
+pub struct IORegister {
+    value: u32,
+    access: IORegisterAccessControl,
+}
+
+impl IORegister {
+    pub fn read(&self) -> u32 {
+        match self.access {
+            IORegisterAccessControl::Read | IORegisterAccessControl::ReadWrite => self.value,
+            _ => panic!("Requested IO Register cannot be read."),
+        }
+    }
+
+    pub fn write(&mut self, value: u32) {
+        match self.access {
+            IORegisterAccessControl::Write | IORegisterAccessControl::ReadWrite => {
+                self.value = value;
+            }
+
+            _ => panic!("Requested IO Register cannot be written."),
+        }
+    }
+
+    pub fn set_byte(&mut self, byte: u8, value: u8) {
+        match self.access {
+            IORegisterAccessControl::Write | IORegisterAccessControl::ReadWrite => {
+                self.value.set_byte(byte, value);
+            }
+
+            _ => panic!("Requested IO Register cannot be written."),
+        }
+    }
+}
+
+#[allow(non_snake_case)]
```

Comment:
Why this?

Reply:
Just follow rust idioms :)

Reply:
In this case I thought that following "CPU" idiom was better but ok I switch to lower case :D

Reply:
You can write whatever you want in comment with `///` for docs

Reply:
I changed to lower-case